### DISCUSSION TITLE
Where can I view console.log from a webpage in a webview?

### DISCUSSION THREAD

**Discussion by oussama-he** [beeware/toga#2354] [Q&A]
Where can I view console.log from a webpage in a webview?

**Comment by mhsmith** [comment]
You should be able to access it using [`evaulate_javascript`](https://toga.readthedocs.io/en/stable/reference/api/widgets/webview.html#toga.WebView.evaluate_javascript).

**Reply by oussama-he** [reply]
Is there a way to debug the JavaScript code? For example, if a page of my app uses JavaScript and I want to debug an unexpected behavior (something like inspector).
Thank you in advance.

**Reply by mhsmith** [reply]
The [documentation](https://toga.readthedocs.io/en/stable/reference/api/widgets/webview.html) contains instructions for enabling the inspector on macOS. 

On other platforms, the best option I can think of is to write some JavaScript that displays debugging information within the page itself, or returns it to Python using `evaluate_javascript`.
